routing/blinding.go

```diff
@@ -86,16 +86,35 @@ func NewBlindedPaymentPathSet(paths []*BlindedPayment) (*BlindedPaymentPathSet,
 		}
 	}
 
-	// NOTE: for now, we just take a single path. By the end of this PR
-	// series, all paths will be kept.
-	path := paths[0]
+	// Derive an ephemeral target priv key that will be injected into each
+	// blinded path final hop.
+	targetPriv, err := btcec.NewPrivateKey()
+	if err != nil {
+		return nil, err
+	}
+	targetPub := targetPriv.PubKey()
```

Comment:
Why do we need to swap this out at all? Doesn't this need to be the same as the `ephKey` that we currently randomly generate when making the invoice in the first place?

Reply:
Is this so even if the blinded paths have a diff pubkey as the final dest, we swap it out to make it look as if they're all going to the same final destination?